charts/tidb-lightning/templates/job.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "tidb-lightning.name" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "tidb-lightning.name" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: tidb-lightning
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "tidb-lightning.name" . }}
+        app.kubernetes.io/instance:  {{ .Release.Name }}
+        app.kubernetes.io/component: tidb-lightning
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path: "/metrics"
+        prometheus.io/port: "8289"
+    spec:
+      {{ if .Values.dataSource.local.nodeName -}}
+      nodeName: {{ .Values.dataSource.local.nodeName }}
+      {{ else -}}
+      initContainers:
+      - name: data-retriever
+        image: {{ .Values.dataSource.remote.rcloneImage }}
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -euo pipefail
+          filename=$(basename {{ .Values.dataSource.remote.path }})
+          rclone --config /etc/rclone/rclone.conf copy -P {{ .Values.dataSource.remote.path }} /data
+          cd /data && tar xzvf ${filename}
+        volumeMounts:
+        - name: credentials
+          mountPath: /etc/rclone
+        - name: data
+          mountPath: /data
+      {{ end -}}
+      restartPolicy: Never      # if lightning fails, manual intervention is required so no restart
+      containers:
+      - name: tidb-lightning
+        image: {{ .Values.image }}
+        imagePullPolicy: {{ .Values.imagePullPolicy | default "IfNotPresent" }}
+        command:
+        - /bin/sh
+        - -c
+        - |
+          data_dir=$(dirname $(find /var/lib/tidb-lightning -name metadata 2>/dev/null) 2>/dev/null)
+          /tidb-lightning \
+          --pd-urls={{ .Values.targetTidbCluster.name }}-pd.{{ .Values.targetTidbCluster.namespace}}:2379 \
+          --status-addr=0.0.0.0:8289 \
+          --importer={{ .Values.targetTidbCluster.name }}-importer.{{ .Values.targetTidbCluster.namespace }}:20170 \
+          --server-mode=false \
+          --tidb-user=root \
+          --tidb-host={{ .Values.targetTidbCluster.name }}-tidb.{{ .Values.targetTidbCluster.namespace }} \
+          {{ if and .Values.dataSource.local.hostPath .Values.dataSource.local.nodeName -}}
+          --d={{ .Values.dataSource.local.hostPath }} \
+          {{ else -}}
+          --d=${data_dir} \
+          {{ end -}}
+          --config=/etc/tidb-lightning/tidb-lightning.toml
+          if [ $? != 0 ]; then
```

Comment:
Currently, there is an error even if the restore succeeds.
```
[2019/08/22 17:58:51.675 +00:00] [INFO] [restore.go:921] ["skip full compaction"]                                                                             
[2019/08/22 17:58:51.675 +00:00] [INFO] [restore.go:435] ["everything imported, stopping periodic actions"]                                                   
[2019/08/22 17:58:51.754 +00:00] [INFO] [restore.go:1073] ["skip clean checkpoints"]                                                                          
[2019/08/22 17:58:51.754 +00:00] [INFO] [restore.go:259] ["the whole procedure completed"] [takeTime=18m48.515158344s] []                                     
[2019/08/22 17:58:51.754 +00:00] [INFO] [main.go:61] ["tidb lightning exit"]
sync log failed sync /dev/stdout: invalid argument
[2019/08/22 17:58:51. tidb-lightning exits abnormally, please exec into my container to do manual intervention
```
So the restore job will not be completed, users have to check the log to verify the status of the restore job.

Reply:
Can we check this in the script?

Reply:
Can we fix tidb-lightning?

Reply:
From @kennytm, it's a known issue https://github.com/uber-go/zap/issues/328

Reply:
Filed pingcap/tidb-lightning#230 to ignore the sync log error.